### DISCUSSION TITLE
Emit event (click) on CButton

### DISCUSSION THREAD

**Discussion by ghost** [coreui/coreui-vue#192] [Q&A]
We are testing out free version of CoreUI 4 for Vue, and are having trouble registering events (for example click) on simple components like CButton. Is there a recommended way to this? Is this maybe only available on PRO version?

Thanks!

**Comment by dang-cb** [comment]
Can you show some code for a CButton that's not working for you? If there's one thing a button *should* be able to do, that's click! 😉

**Reply by ghost** [reply]
for example, we tried adding @click event listener on CButton, but click event didn't register:
```vue
<c-button
    @click="login"
    :disabled="isLoading"
    color="primary"
    type="submit"
    class="px-4"
>
    Login
</c-button>
```

If we add a wrapper div around it, and place the @click event listener there, it registers the click.

**Comment by mrholek** [comment]
@aviatmcel I will try to release the fix tomorrow.

BTW. I replied to your support ticket 2 days ago, did you get my email?

**Reply by ghost** [reply]
Thanks, appreciate it.

Hmm no, I didn't get your email, not in spam folder either, will check with our IT department. I posted my question on https://coreui.io/support/

**Reply by mrholek** [reply]
@aviatmcel which version of @coreui/vue-pro do you use, I tried to reproduce this issue, but everything is ok on my side.

**Reply by ghost** [reply]
Hmm, maybe its because we were using the non-pro version?
"@coreui/vue": "^4.1.2",

**Reply by mrholek** [reply]
Can you try to remove this line `:disabled="isLoading"` and check.

If disabled is to true then the button doesn't emit the event.

**Reply by ghost** [reply]
hmm no, removing ` :disabled="isLoading"` also didn't help.

I can make it work by adding .native to @click, like this:
`<c-button @click.native="login">Login CB</c-button>`

Maybe I forgot to import some part of CoreUI lib? I have these imports in **main.js**:
```
import '@coreui/coreui/dist/css/coreui.min.css';
import CoreUI from '@coreui/vue';
app.use(CoreUI);
```

BTW, still not getting any email response from support :)

**Reply by mrholek** [reply]
Please change `import CoreUI from '@coreui/vue';` => `import CoreuiVue from '@coreui/vue'`, and `app.use(CoreUI);` => `app.use(CoreuiVue);`

**Reply by ghost** [reply]
hmm, changed this, but its still the same, only works with @click.**native**

**Reply by mrholek** [reply]
@aviatmcel please update to version v4.2.0
